snuba_sdk/column.py

```diff
@@ -32,6 +32,7 @@ class Column(Expression):
 
     name: str
     entity: Optional[Entity] = None
+    output_alias: Optional[str] = None
```

Comment:
I would leave a comment here explaining what an output alias is as it is not very obvious

Reply:
Also, how should it show up in the string representation of the expression?

Reply:
You can use columns everywhere in the query. While the output_alias would only be allowed for columns that are in the top level select clause and top level group by. This makes me think that the place for such alias is not a field in the column but another data structure like we do for the query object in snuba.

Why don't we change the type of the select clause and the group by clause from a sequence of Expresison to a Serquence of AliasedExpressions (equivalent of SelectedExpression in the snuba AST). Where AliasedExpression contains an expression and provides the alias. 

That is supposed to make the usage of the Column class structurally sound (you cannot structurally add an alias in clauses where that is forbidden without having to add additional validation at runtime). And you do not need to manage the output alias separately in all types of expressions you can have in the select clause.

Reply:
I added a comment about what an output alias is for. It shows up in the query as `name AS alias`.

Reply:
The reason I did things this way was to make it easier to reuse Expressions throughout queries. Having this field be in the Column class but optional allows users to reuse the Column everywhere in the Query object without having to redefine it or wrap it in a class.

I'll try with the AliasedExpression and see if that works in a better way.